### PR TITLE
Add DeDoDe (clean version)

### PR DESCRIPTION
**Pull request by ducha-aiki**
#### Changes
Copied version of https://github.com/kornia/kornia/pull/2787, with fixes, also starting from `main` batch

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [x] 🔬 New feature (non-breaking change which adds functionality)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by ducha-aiki** [comment]
@edgarriba @johnnv1 I believe the following checks are unnecessary strict, especially given high readability of DiNOv2 code. 

image

**Comment by johnnv1** [comment]
> @edgarriba @johnnv1 I believe the following checks are unnecessary strict, especially given high readability of DiNOv2 code.

Would say to just add noqa: <rule number> on these lines

**Review by edgarriba — CHANGES_REQUESTED** [review]
For now only the potential cyclic issue. We have trying to isolate in the top level augmentations module (treat as an external library). The rest future work. Besides, there’s a large chunk of uncovered code from the model code. Verify if it’s all useful otherwise add as much tests as possible

File: kornia/feature/dedode/dedode.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
+from typing import Dict, Optional, Tuple
+
+import torch
+import torch.nn.functional as F
+
+from kornia.augmentation._2d.intensity.normalize import Normalize
```
[the remark this answers is not in the document]
Comment (ducha-aiki) [reply kornia/feature/dedode/dedode.py:6]: The reason I used this, to avoid creating the mean-std tensor on device every time. But yes, circulars are much more important, I will use normalize from enhance for now

File: kornia/feature/dedode/dedode.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    """
+
+    # TODO: implement steerers and mnn matchers
+    def __init__(
+        self, detector_model: str = "L", descriptor_model: str = "G", amp_dtype: torch.dtype = torch.float16
+    ) -> None:
+        super().__init__()
+        self.detector: DeDoDeDetector = get_detector(detector_model, amp_dtype)
+        self.descriptor: DeDoDeDescriptor = get_descriptor(descriptor_model, amp_dtype)
+        self.normalizer = Normalize(torch.tensor([0.485, 0.456, 0.406]), std=torch.tensor([0.229, 0.224, 0.225]))
```
[the remark this answers is not in the document]
Comment (ducha-aiki) [reply kornia/feature/dedode/dedode.py:52]: @edgarriba actually we do have it already with exact same functionality, as in augmentation. 
It is just the docs search showed me only the augmentation one

File: kornia/feature/dedode/transformer/dinov2.py
[the remark this answers is not in the document]
Comment (ducha-aiki) [reply kornia/feature/dedode/transformer/dinov2.py:1]: I created an issue here https://github.com/kornia/kornia/issues/2841

File: kornia/feature/dedode/vgg.py
[the remark this answers is not in the document]
Comment (ducha-aiki) [reply kornia/feature/dedode/vgg.py:1]: https://github.com/kornia/kornia/issues/2841

File: kornia/feature/dedode/dedode.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (edgarriba) [reply kornia/feature/dedode/dedode.py]: I advocate for that utility class I mentioned or for next pr

[the remark this answers is not in the document]
Comment (ducha-aiki) [reply kornia/feature/dedode/dedode.py]: @edgarriba as I said in another comment - the utility class is already there %)

File: kornia/feature/dedode/dedode.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (edgarriba) [reply kornia/feature/dedode/dedode.py:52]: nice
